src/org/daisy/dotify/formatter/impl/volume/TocSequenceEventImpl.java

```diff
@@ -121,16 +123,23 @@ public BlockSequence getBlockSequence(FormatterContext context, DefaultContext v
 					getSequenceProperties());
 			fsm.appendGroup(getTocStart(vars));
 			if (getRange()==TocProperties.TocRange.VOLUME) {
-				Collection<Block> volumeToc = data.filter(refToVolume(vars.getCurrentVolume(), crh));
+                int currentVolume = vars.getCurrentVolume();
+                Collection<Block> resumedBlocks = data.filterEntryOnResumed(rangeToVolume(currentVolume, crh));
+				Collection<Block> volumeToc = data.filterEntry(refToVolume(currentVolume, crh));
+                if (resumedBlocks.isEmpty() && volumeToc.isEmpty()) {
+					return null;
+				}
+                if (!resumedBlocks.isEmpty()) {
+                    fsm.appendGroup(resumedBlocks);
+                }
 				if (!volumeToc.isEmpty()) {
 					fsm.appendGroup(volumeToc);
-				} else {
-					return null;
```

Comment:
Is it safe to continue execution if the volumeToc is empty?

Reply:
Yes, this was originally designed so. In other words, before we started with the `<toc-block>` construction, also `null` was returned when the TOC was empty.